tower/src/builder/service.rs

```diff
@@ -0,0 +1,79 @@
+use super::Error;
+use futures::{Async, Future, Poll};
+use std::marker::PhantomData;
+use std::sync::Arc;
+use tower_layer::Layer;
+use tower_service_util::MakeService;
+use Service;
+
+/// Composed `MakeService` produced from `ServiceBuilder`
+#[derive(Debug)]
+pub struct ServiceBuilderMaker<S, L, Request> {
+    maker: S,
+    layer: Arc<L>,
+    _pd: PhantomData<Request>,
+}
+
+/// Async resolve the MakeService and wrap it with the layers
+#[derive(Debug)]
+pub struct MakerFuture<S, L, Target, Request>
```

Comment:
We've been calling the future type `[Item]Future`. Maybe this should be `ServiceFuture`?